webapp/src/registerDog/registerDog.tsx

```diff
@@ -107,9 +107,10 @@ export default function RegisterDogForm() {
     sessionStorage.setItem("inputField", JSON.stringify(newField));
   };
 
-  const onSavePostClicked = () => {
+  const onSubmitClicked = () => {
     try {
       registerDog(lostDogFields, picture);
```

Comment:
I think upong dog registration, you should set refreshRequired to true in reducer, so that when we are redirected to doglist, we already see the updated list

Reply:
The list is refreshed, but it fetches 15 dogs from starting from the oldest ones so that's why the dog that was added is not displayed immediately.